clique/src/service.rs

```diff
@@ -85,6 +92,41 @@ pub trait SpawnHandleT {
     ) -> Pin<Box<dyn Future<Output = Result<(), ()>> + Send>>;
 }
 
+#[derive(Debug)]
+pub enum CliqueNetworkServiceError {
+    Commands,
+    AuthorizationRequests,
+    ConnectionWorker,
+}
+
+impl Display for CliqueNetworkServiceError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CliqueNetworkServiceError::Commands => write!(f, "Stream with commands was closed."),
+            CliqueNetworkServiceError::AuthorizationRequests => {
+                write!(f, "`authorization_requests` channel was closed.")
+            }
+            CliqueNetworkServiceError::ConnectionWorker => {
+                write!(f, "`worker_results` channel was closed.")
+            }
+        }
+    }
+}
+
+impl CliqueNetworkServiceError {
+    fn commands_stream_closed() -> Self {
```

Comment:
These constructors don't really seem necessary, no?

Reply:
:D

Reply:
I mean, you can just use `enum` variants as constructors as we do in all other code, right? <_<"

Reply:
I thought the goal was to make it future-proof. This way we can add more details to them later more easily (e.g. back-traces). Just kidding ;)